src/main/java/com/pingcap/tikv/expression/TiConstant.java

```diff
@@ -78,6 +82,22 @@ public Expr toProto() {
     } else if (value instanceof BigDecimal) {
       builder.setTp(ExprType.MysqlDecimal);
       DecimalType.writeDecimal(cdo, (BigDecimal) value);
+    } else if (value instanceof java.sql.Date) {
+      builder.setTp(ExprType.MysqlTime);
+      Calendar calendar = Calendar.getInstance();
```

Comment:
When you construct a calendar like this, it means we choose to use default timezone. And correspondingly, when you use calendar to extract date time, it involves timezone shift.
Please confirm if this is what you want.

Reply:
Done